src/boot/memory.c

```diff
@@ -271,6 +272,52 @@ void dma_read(u8 *dest, u8 *srcStart, u8 *srcEnd) {
     }
 }
 
+struct DMAAsyncCtx {
+    u8* srcStart;
+    u8* dest;
+    u32 size;
+};
+
+// Starts to DMA the first block
+static void dma_async_ctx_init(struct DMAAsyncCtx* ctx, u8 *dest, u8 *srcStart, u8 *srcEnd) {
```

Comment:
move lz4t specific stuff to its own c file

Reply:
Ideally we should move compression libraries to the lib folder or something so they're all in one place. Personally not huge on dropping these into their own src folders or inside the boot folder, where they don't really belong.

Reply:
Currently kept it in boot. It is planned to do a complete overhaul by removing YAZ0, MIO0, RNC* algos and adding Shrinkler/aplib. I think it would be a good idea to consider moving files at that time.